src/test/regress/sql/multi_partitioning.sql

```diff
@@ -1162,6 +1162,47 @@ DROP TABLE partitioning_test, partitioning_test_2008, partitioning_test_2009,
            partitioning_test_2010, partitioning_test_2011,
            reference_table, reference_table_2;
 
+-- test cases for #3970
```

Comment:
This test has all it needs in itself, I think it would make sense to create a separate file for this test with a name `partitioning_issue_3970.sql` Why?:

- Because then this is more isolated, doesn't pollute this file unnecessarily
- This test contains metadata dependent stuff, such as we connect to the worker 1 and do some verification. In the future, if we want to have metadata independent tests, it will make things easier.